compiler/backend/src/org/jetbrains/kotlin/codegen/inline/SMAP.kt

```diff
@@ -39,25 +40,25 @@ class SMAPBuilder(
 
         val defaultStrata = generateDefaultStrata(realMappings)
         val debugStrata = generateDebugStrata(realMappings)
-
-        return "$header\n$defaultStrata$debugStrata"
+        if (backwardsCompatibleSyntax && defaultStrata.isNotEmpty() && debugStrata.isNotEmpty()) {
+            // Old versions of kotlinc might fail if there is no END between defaultStrata and debugStrata.
+            // This is not actually correct syntax according to JSR-045.
+            return "$header\n$defaultStrata$END\n$debugStrata$END\n"
+        }
+        return "$header\n$defaultStrata$debugStrata$END\n"
```

Comment:
@udalov Looks like it's also best to add message from future for classes with strata in new format

Reply:
I suppose we should make the new language feature's kind `UNSTABLE_FEATURE`, so that manually enabling it would produce pre-release binaries. But otherwise we need not do anything, since 1.5 binaries will only be readable by 1.4+.